addon/instance-initializers/setup-session-restoration.js

```diff
@@ -1,13 +1,14 @@
-import Ember from 'ember';
-
 export default function setupSessionRestoration(instance) {
   const { container } = instance;
   const applicationRoute = container.lookup('route:application');
   const session = container.lookup('simple-auth-session:main');
   applicationRoute.reopen({
-    beforeModel(transition) {
-      const superCall = Ember.run.bind(this, this._super, transition);
-      return session.restore().then(superCall, superCall);
+    beforeModel() {
+      const superResult = this._super(...arguments);
+      const superReturn = function() {
+        return superResult;
+      };
+      return session.restore().then(superReturn, superReturn);
```

Comment:
If the `session.restore()` promise is returned from beforeModel(), and the session restore promise is rejected, the transition errors and does not complete. This means finally cannot be used.

`return session.restore().then(() => superResult, () => superResult);` works.